### ISSUE TITLE
Crash on 0.8.4 Sierra when activating new local file

### ISSUE THREAD

**Issue by SilverPin** [2ndalpha/gasmask#113]
As it says in the title: when I click activate on newly created local file, Gasmask crashes.

**Comment by MadMacMad** [comment]
same here...
also activating preferences crash...

**Comment by thesadboy** [comment]
same here...

**Comment by rcrooks** [comment]
and same here, everytime I try to switch hosts

**Comment by webfor-alan** [comment]
Samsies, every time I try to switch hosts.

**Comment by andoro** [comment]
The same here, but it doesn't have to be local file, it crashes everytime when you change your hosts file. I don't really understand this, because last week it worked for me, I'm used Sierra that time too.
I've attached the crash log, I hope it helps for the fix.
[gasmask_crash.txt](https://github.com/2ndalpha/gasmask/files/636780/gasmask_crash.txt)

**Comment by rcrooks** [comment]
I think I may have found a solution: run this in Terminal:

sudo /bin/chmod +a 'user:YOUR_USERNAME:allow write' /private/etc/hosts

Worked for me at least -- Gasmask no longer crashing.

**Comment by SilverPin** [comment]
Thanks for the workaround. I'm not sufficiently competent with command line interface to know what that workaround does exactly. I just downgraded to 0.8.3 and skipped the 0.8.4 update I was offered. Works like a charm. 

I am pleased you found a work around that doesn't involve downloading a fresh copy.

**Comment by rcrooks** [comment]
my change just gives you write permissions on the hosts directory. It's odd that 0.8.3 works but 0.8.4 doesn't, if that's the issue....

**Comment by emenoh** [comment]
running the permission fix works for me.

**Comment by emenoh** [comment]
@emenoh wahhhh

**Comment by thesadboy** [comment]
permission fix works fine for me, thanks a lot.

**Comment by tomasantonj** [comment]
Permission change worked for me too.

**Comment by emenoh** [comment]
Permission change does not persist, have to rerun the command. Not sure about causes. Running VPN sometimes, running Parallels always... encrypted disk, so YMMV.

**Comment by darknate** [comment]
allowing write helped me to, thx!

**Comment by alanmcginnis** [comment]
Permission change worked for me too.

**Comment by andrearufo** [comment]
`sudo /bin/chmod +a 'user:YOUR_USERNAME:allow write' /private/etc/hosts`

It works for me!

**Comment by thesadboy** [comment]
Permission change can only work for short time...
And then crash again.

**Comment by SilverPin** [comment]
The earlier release is still working perfectly for me - hasn't crashed once. 

I said earlier it was 0.8.3, but I realise now it is actually 0.8.1. You can download it here:

https://github.com/2ndalpha/gasmask/releases

**Comment by oliveratgithub** [comment]
Same issue here - workaround by @rcrooks worked like a charm, thanks!

**Comment by keoshi** [comment]
You're a lifesaver, @rcrooks — thank you!

**Comment by lifeibest** [comment]
@rcrooks It's works. My os is 10.12, work well.

**Comment by rcrooks** [comment]
Glad the permissions fix seems to be working for everyone - I've been sharing it with colleagues here and they've found it to work also. I got the idea because when I started running Sierra betas, gasmask started prompting me for a system password sometimes when I switched hosts. It would be nice if this fix could be incorporated into the app, or at least a warning with instructions on what to do.

@thesadboy not sure why it isn't sticking for you. Are you by chance running some kind of process (maybe embedded in some system maintenance tool) that restores permissions to their original state?
